src/engine/input/key-mapping/InputEventWithModifiersConverter.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using Godot;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+public class InputEventWithModifiersConverter : JsonConverter<InputEventWithModifiers>
```

Comment:
I'm a bit worried that as this creates Godot.Reference inheriting objects, this can permanently leak objects.
See my previous comment about using a custom class instead than InputEventWithModifiers.

Reply:
The Node based deserializer in many cases will register the deserialized objects to be freed later.

Reply:
But I do require a InputEventWithModifiers at some point for passing it to the InputMap class.
I thought that working with one class only would lead to better code, but if this really produces memory leaks I will change this

Reply:
On reviewing the Reference class again, it kinda seems it doesn't. But still that class is in Godot and implemented in C++, meaning that marshalling etc. call overhead from C# is incurred whenever it is used.
If a separate class is created, it's obvious that it needs a method to create an equivalent InputEventWithModifiers for use elsewhere.